### PR TITLE
Add `MBart` support for `BetterTransformer`

### PR DESCRIPTION
**Pull request by ravenouse**
This PR adds the `MBartEncoderLayerBetterTransformer` class to support `MBart`.

During testing, the code is runnable but it yields different results compared with the original transformer model, as shown in the below picture. The specific models I tested are `facebook/mbart-large-50` and `facebook/mbart-large-cc25`. The downstream task I tested is fill masks

截屏2022-11-26 下午2 17 04
截屏2022-11-26 下午2 17 19

Could you tell me what I can do to solve this problem and what other tests I need to run?

Thank you so much!

### CODE REVIEW

File: optimum/bettertransformer/models/encoder_models.py [hunk 1]
```diff
@@ -343,6 +343,114 @@ def forward(self, hidden_states, attention_mask, position_bias=None, *_, **__):
...
+        self.norm2_weight = mbart_layer.final_layer_norm.weight
+        self.norm2_bias = mbart_layer.final_layer_norm.bias
+
+        # Model hyper parameters
+        self.num_heads = mbart_layer.self_attn.num_heads
+        self.embed_dim = mbart_layer.self_attn.embed_dim
+
+        # Last step: set the last layer to `False` -> this will be set to `True` when converting the model
+        self.is_last_layer = False
+
```
Comment (younesbelkada) [inline optimum/bettertransformer/models/encoder_models.py:404]:
```suggestion
        self.norm_first = True
```
As you can see [here](https://github.com/huggingface/transformers/blob/95754b47a6d4fbdad3440a45762531e8c471c528/src/transformers/models/mbart/modeling_mbart.py#L332), `MBart` seems to use pre-attention layer norm. Setting this attribute to `True` should fix the issue

**Review by younesbelkada — COMMENTED** [review]
Hi @ravenouse !
Thanks a lot for your PR! Glad that the conversion worked! We're almost there!
It seems that `MBart` uses pre-attention layer norm, could you try to set the attribute to `True` (as suggested on the suggestion below)
Let me know how it goes!

**Comment by younesbelkada** [comment]
Very glad it worked @ravenouse ! Can't wait to see this PR to be merged!
Let me guide you step by step regarding the immediate next steps:

Step 1: Finish the integration for `MBart`
- Since the refactoring for encoder-decoder based models tests has been recently addressed, adding the tests for MBart should be very easy. Just add a line with `"hf-internal-testing/tiny-random-MBartModel"` on the list `ALL_ENCODER_DECODER_MODELS_TESTS` [here](https://github.com/ravenouse/optimum/blob/478473f4c7950a15a8a6214ec6b17755561224eb/tests/bettertransformer/test_bettertransformer_encoder.py#L51) - push this change to see if the test pass (it should pass if everything works as expected , or you can just run the command `pytest tests/bettertransformer/test_bettertransformer_encoder.py::BetterTransformersEncoderDecoderTest`
- Add a line stating that `MBart` is supported for `BetterTransformer` on the documentation. The addition should go [here](https://github.com/ravenouse/optimum/blob/478473f4c7950a15a8a6214ec6b17755561224eb/docs/source/bettertransformer/overview.mdx?plain=1#L40) (make sure to respect the alphabetical order)

Step 2: Add the integration for `NLLB`
- `NLLB` is a recent encoder-decoder model from Meta AI that is integrated in `transformers`. Since `NLLB` uses **the same** encoder layer as `MBart` (see [this line](https://github.com/huggingface/transformers/blob/95754b47a6d4fbdad3440a45762531e8c471c528/src/transformers/models/m2m_100/modeling_m2m_100.py#L344)) the addition is very easy. Just add the line `"M2M100EncoderLayer":MBartEncoderLayerBetterTransformer` [here](https://github.com/ravenouse/optimum/blob/478473f4c7950a15a8a6214ec6b17755561224eb/optimum/bettertransformer/models/__init__.py#L30).
- Once the model is integrated as follows. Follow the same procedure as *Step 1*, and you can use the model `hf-internal-testing/tiny-random-nllb` for the integration tests ;) 
Let me know if anything is unclear! Again thanks a lot for your great contribution @ravenouse 💪

File: tests/bettertransformer/test_bettertransformer_encoder.py [hunk 2]
```diff
@@ -44,6 +44,8 @@
     "hf-internal-testing/tiny-random-MarkupLMModel",
     "hf-internal-testing/tiny-random-BertModel",
     "ybelkada/random-tiny-BertGenerationModel",
+    "hf-internal-testing/tiny-random-MBartModel",
+    "hf-internal-testing/tiny-random-nllb",
```
Comment (younesbelkada) [inline tests/bettertransformer/test_bettertransformer_encoder.py:48]: I think that you have to move them to `ALL_ENCODER_DECODER_MODELS`: the test `pytest tests/bettertransformer/test_bettertransformer_encoder.py::BetterTransformersEncoderDecoderTest` will only run on the models listed on `ALL_ENCODER_DECODER_MODELS` ;)

**Review by younesbelkada — COMMENTED** [review]
Thanks so much @ravenouse !
I left a small comment, could you please update `ALL_ENCODER_DECODER_MODELS` instead so that the tests will run on the new models? Of course let me know how it goes!

**Comment by younesbelkada** [comment]
Thanks a lot! 
Could you just run the styling (run "make style") push the changes and we should be good to merge! 💪🏻

**Review by younesbelkada — APPROVED** [review]
Looking great! 
Thanks a lot for your clean implementation of `M2M100` support for `BetterTransformer` 💪 
Looking forward to your next contributions ;)

**Review by michaelbenayoun — APPROVED** [review]
LGTM!

**Comment by younesbelkada** [comment]
Thanks so much @ravenouse !
Sure yes, just make sure to notify other contributors in https://github.com/huggingface/transformers/issues/20372 too ;)
